src/Lean/Elab/Deriving/BEq.lean

```diff
@@ -97,15 +97,34 @@ private def mkBEqInstanceCmds (declNames : Array Name) : TermElabM (Array Syntax
   trace[Elab.Deriving.beq] "\n{cmds}"
   return cmds
 
+private def mkBEqEnumFun (ctx : Context) (name : Name) : TermElabM Syntax := do
+  let auxFunName ← ctx.auxFunNames[0]
+  let indVal     ← ctx.typeInfos[0]
+  let x_ ← mkFreshUserName `x
+  let y_ ← mkFreshUserName `y
+  let body ← `($(mkIdent x_).toCtorIdx == $(mkIdent y_).toCtorIdx)
+  `(private def $(mkIdent auxFunName):ident ($(mkIdent x_):ident : $(mkIdent name)) ($(mkIdent y_):ident : $(mkIdent name)) : Bool := $body:term)
```

Comment:
Lean macros are hygienic. You can write
```lean
  `(private def $(mkIdent auxFunName):ident  (x y : $(mkIdent name)) : Bool := x.toCtorIdx == y.toCtorIdx)
```
There is no risk of accidental name capture.